s3/src/main/scala/akka/stream/alpakka/s3/impl/HttpRequests.scala

```diff
@@ -11,12 +11,35 @@ import akka.http.scaladsl.model.{ContentTypes, RequestEntity, _}
 import akka.stream.alpakka.s3.S3Settings
 import akka.stream.scaladsl.Source
 import akka.util.ByteString
-
 import scala.collection.immutable.Seq
 import scala.concurrent.{ExecutionContext, Future}
 
 private[alpakka] object HttpRequests {
 
+  def listBucket(
+      bucket: String,
+      region: String,
+      prefix: Option[String] = None,
+      continuation_token: Option[String] = None
+  )(implicit conf: S3Settings): HttpRequest = {
+
+    val listTypeQuery: (String, String) = "list-type" -> "2"
```

Comment:
Ya, this is a little nicer. Still leveling up on my use of the collections in scala and using them properly.